Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py

```diff
@@ -914,15 +914,18 @@ def _to_git_ref(self, argument):
         if not isinstance(argument, six.string_types):
             raise ValueError("Expected 'argument' to be a string, not '{}'".format(type(argument)))
         parsed_commit = Commit.parse(argument, do_assert=False)
-        if parsed_commit and not parsed_commit.hash:
-            return self.commit(
-                hash=parsed_commit.hash,
-                revision=parsed_commit.revision,
-                identifier=parsed_commit.identifier,
-                branch=parsed_commit.branch,
-                include_log=False,
-                include_identifier=False,
-            ).hash
+        try:
+            if parsed_commit and not parsed_commit.hash:
+                return self.commit(
+                    hash=parsed_commit.hash,
+                    revision=parsed_commit.revision,
+                    identifier=parsed_commit.identifier,
+                    branch=parsed_commit.branch,
+                    include_log=False,
+                    include_identifier=False,
+                ).hash
+        except self.Exception:
+            pass
```

Comment:
Would appreciate a comment here explaining why we're swallowing exceptions like this. And if it's possible to narrow the scope down from _every_ git-related exception, that'd be great.

Reply:
I can add the comment, but basically, we actually _want_ to swallow every git-related exception.

The name of this function is `_to_git_ref`. The assumption here is that the user could be specifying any number of things which are convertible to a `git` ref. Most likely, it's an identifier, revision (old SVN style revision, so `r1234`) or a hash, but it could also a branch name. It's that later one that can be challenging, because something like `12345` is a commit hash (or possible a commit hash), but could also be used as a branch name. The trouble with a string argument like `12345` is that until we start fiddling with it, we have no way of knowing which it is. This code will assume that `12345` is a hash, and attempt to construct a commit. If it is, in fact, a hash, we'll construct a commit and return the full hash. If it's not a hash, though, then `self.commit` is going to raise an exception, and we should instruct our caller to assume that `12345` must be a branch name. This is rarely an issue in practice because we strongly encourage folks to prefix branches with `eng/`, and such a branch is clearly not a hash.

Reply:
I see. So, the purpose of this argument is really to transform a commit identifier down into something that can be handed down to git, and if anything other than an identifier is given, we can safely pass it on without parsing. I wonder if we could be more explicit about that (i.e. taking the `self.commit` branch IFF `argument` is a commit identifier), but a comment explaining why it's okay to return an unprocessed `argument` is fine with me too.